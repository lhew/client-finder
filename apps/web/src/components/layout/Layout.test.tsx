import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Layout } from ".";

describe("<Layout />", () => {
  it("renders correctly in default mode", () => {
    const { container } = render(
      <Layout type="splash" withNavbar={false}>
        layout content - slash screen
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correctly for standard layout (internal pages ,etc)", () => {
    const { container } = render(
      <Layout>layout content - internal pages </Layout>
    );
    expect(container).toMatchSnapshot();
  });
});
