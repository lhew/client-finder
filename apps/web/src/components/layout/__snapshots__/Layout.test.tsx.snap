// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders correctly for standard layout (internal pages ,etc) 1`] = `
<div>
  <div
    class="min-h-[100vh] w-full"
  >
    <div
      class="fixed  top-0 w-full border-b-2 border-gray-400 bg-white"
      data-testid="layout-navbar"
    >
      <div
        class="align-center mx-[auto] grid max-w-5xl grid-cols-[1fr_20em] px-2 py-4"
      >
        <h1
          class="font-bold"
        >
          Client Finder
        </h1>
      </div>
    </div>
    <div
      class="mx-[auto] flex min-h-[100vh] flex-col max-w-5xl pt-[5em]"
    >
      <div
        class="flex-1 px-2"
      >
        layout content - internal pages 
      </div>
      <footer
        class="p-3 text-center text-xs text-gray-500"
      >
        Made by Leonardo Almeida
      </footer>
    </div>
  </div>
</div>
`;

exports[`<Layout /> renders correctly in default mode 1`] = `
<div>
  <div
    class="mx-auto grid h-[100vh] grid-rows-[1fr_20em]"
  >
    <div
      class="mx-[auto] flex min-h-[100vh] flex-col w-3/5"
    >
      <div
        class="flex-1 px-2"
      >
        layout content - slash screen
      </div>
      <footer
        class="p-3 text-center text-xs text-gray-500"
      >
        Made by Leonardo Almeida
      </footer>
    </div>
  </div>
</div>
`;
